### ISSUE TITLE
PAL doesn't add data for Process/Working Set Private to the htm report

### ISSUE THREAD

**Issue by SDaniV** [clinthuffman/PAL#35]
Hello Clint,

I encounter a problem that PAL doesn't generate data for Process/Working Set Private counter in htm report. When my Performance Counter contains many counters it looks like PAL just ignore Process/Working Set Private and generate the report without it.
I created empty Performance Counter NewDataCollector, added only Process/Working Set Private there and I got such error message:

Overall progress... Status: Calculating counter statistics [very CPU intensive]..., PAL 2.8.1 Progress: 65%... Calculating counter statistics [very CPU intensive]..., 

An error occurred on...
$iPercentComplete = ConvertToDataType $(($iIndexOfJobs / $iTotalJobs) * 100) 'integer' 
At C:\Program Files\PAL\PAL\PAL.ps1:2526 char:49 + ... ete = ConvertToDataType $(($iIndexOfJobs / $iTotalJobs) * 100) 'integ ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Attempted to divide by zero.

SCRIPT ARGUMENTS:
Log: C:\PerfLogs\Admin\NewDataCollector\BLACKBIRD_20190204-000001\DataCollector01.blg
ThresholdFile: C:\Program Files\PAL\PAL\SystemOverview.xml
AnalysisInterval: AUTO
IsOutputHtml: $True
IsOutputXml: $False
HtmlOutputFileName: [LogFileName]_PAL_ANALYSIS_[DateTimeStamp].htm
XmlOutputFileName: [LogFileName]_PAL_ANALYSIS_[DateTimeStamp].xml
OutputDir: D:\PAL_results\IDLE_new_test
AllCounterStats: $False
NumberOfThreads: 4
IsLowPriority: $True
DisplayReport: True

However, Working Set Private exists in the .blg file.

I use PAL v2.8.1 on Windows 10 64-bit.

**Comment by jeffstokes72** [comment]
I was curious on the applicability of this as well, I have to admit...

**Comment by SDaniV** [comment]
Hi Clint,

I'm sorry that it took so long to get back to you.

I'm just interested in the statistics to compare Working Set with Working Set - Private for a specific process and add it to .htm report. I've been trying to enable All Counter Stats checkbox but the counter wasn't included to the report..

Is it possible to modify the current file without creating a new one? When I go to Threshold file -> Edit -> New ...(add counter) -> Update Analysis -> Save I'm getting error message.

Also, I tried to create a new one as you described in the article but for me not everything is clear:
Should I modify the tool to use it instead of SystemOverview.xml or there is some easiest way :)

Thank you

**Comment by clinthuffman** [comment]
Likely you can’t save the file because it is in a secure location. Try copying the PAL installation folder to your desktop and then running PALWizard.exe from there, then try editing and saving the threshold file.

There again, if you are just comparing two counters, then viewing them in Perfmon sounds easier.

Thank you,


________________________________

**Comment by SDaniV** [comment]
Hi Clint,

I'm terribly sorry, I tried your suggestion only now.
When I try to edit System Overview I get Working Set - Private to the report but without the chart and for some reason, half of the other charts break. At the same time, when I try to edit Quick System Overview everything works fine but in such case, I can't get another process counters...

Thank you for trying to help me!

**Comment by clinthuffman** [comment]
[SystemOverview.xml.txt](https://github.com/clinthuffman/PAL/files/2870480/SystemOverview.xml.txt)
Hi SDaniV, I updated SystemOverview.xml with the Working Set - Private counter. It is statistics only - no thresholds, but it works. I attached it to this response. 

Rename SystemOverview.xml.txt (attached) to SystemOverview.xml and then replace the same named file in the PAL installation folder, then try it out.
